multicluster/controllers/multicluster/commonarea/remote_common_area.go

```diff
@@ -206,29 +207,27 @@ func getSecretCACrtAndToken(secretObj *v1.Secret) ([]byte, []byte, error) {
 }
 
 func (r *remoteCommonArea) SendMemberAnnounce() error {
-	memberAnnounceList := &multiclusterv1alpha1.MemberClusterAnnounceList{}
-	if err := r.List(context.TODO(), memberAnnounceList, client.InNamespace(r.GetNamespace())); err != nil {
+	var err error
+	memberAnnounceName := "member-announce-from-" + r.GetLocalClusterID()
+	existingMemberAnnounce := &multiclusterv1alpha1.MemberClusterAnnounce{}
+	if err = r.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.GetNamespace(),
+		Name:      memberAnnounceName,
+	}, existingMemberAnnounce); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
+	var localClusterMemberAnnounceExist bool
 	var localClusterMemberAnnounce multiclusterv1alpha1.MemberClusterAnnounce
-	localClusterMemberAnnounceExist := false
-	if len(memberAnnounceList.Items) != 0 {
-		for _, memberAnnounce := range memberAnnounceList.Items {
-			if memberAnnounce.ClusterID == r.GetLocalClusterID() {
-				localClusterMemberAnnounceExist = true
-				localClusterMemberAnnounce = memberAnnounce
-				break
-			}
-		}
+
+	if apierrors.IsNotFound(err) {
```

Comment:
We can just do:

localClusterMemberAnnounceExists := err == nil

Reply:
Done